### PR TITLE
feat: adding create bot functionality

### PR DESCRIPTION
**Pull request by namchuai**
Fixes: #232 
- [x] Add toggling for Chat history (with animation)
- [x] Add UI for creating bot
- [x] Map function

Screenshot 2023-10-16 at 18 20 15
Screenshot 2023-10-16 at 18 17 29

### CODE REVIEW

**Comment by dan-jan** [comment]
@namchuai Looking good!

**Comment by 0xSage** [comment]
so cool!

File: plugins/data-plugin/index.ts [hunk 1]
```diff
@@ -154,6 +155,11 @@ export function init({ register }: { register: RegisterExtensionPoint }) {
   register(DataService.DeleteConversation, deleteConversation.name, deleteConversation);
   register(DataService.CreateMessage, createMessage.name, createMessage);
   register(DataService.GetConversationMessages, getConversationMessages.name, getConversationMessages);
+
+  register("createBot", createBot.name, createBot);
```
Comment (namchuai) [inline plugins/data-plugin/index.ts:159]: Self note: Define in `DataService` and publish new version on to npm

**Review by louis-jan — APPROVED** [review]
Lgtm

**Review by urmauur — APPROVED** [review]

File: web/app/_components/ToggleSwitch/index.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import { Switch } from "@headlessui/react";
+import { Controller } from "react-hook-form";
+
+function classNames(...classes: any) {
```
Comment (louis-jan) [inline web/app/_components/ToggleSwitch/index.tsx:5]: Can import classnames, no need to redefine

**Comment by louis-jan** [comment]
I will still proceed & merge, @urmauur will help on this

Screenshot 2023-10-23 at 15 56 24

**Pull request merged**